src/htsp_server.c

```diff
@@ -906,6 +906,10 @@ htsp_build_channel(channel_t *ch, const char *method, htsp_connection_t *htsp)
     if (t->s_hbbtv)
       htsmsg_add_msg(svcmsg, "hbbtv", htsmsg_copy(t->s_hbbtv));
 
+    /* Provider */
+    if (t->s_provider_name)
+      htsmsg_add_str(svcmsg, "providerName", t->s_provider_name);
```

Comment:
You can use `htsmsg_add_str2` here, which already includes the null check (great naming of that function, I know, that is also on the todo list)

Reply:
Thanks. Will do.